sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -407,13 +407,24 @@ case class GpuSortArray(base: Expression, ascendingOrder: Expression)
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): cudf.ColumnVector = {
     val isDescending = isDescendingOrder(rhs)
-    lhs.getBase.listSortRows(isDescending, true)
+    val base = lhs.getBase()
+    dataType match {
+      case ArrayType(FloatType | DoubleType, _) =>
+        withResource(base.getChildColumnView(0)) {child =>
+          withResource(child.normalizeNANsAndZeros()) {normalizedChild =>
+            withResource(base.replaceListChild(normalizedChild)) {normalizedList =>
+              normalizedList.listSortRows(isDescending, true)
+            }
+          }
+        }
+      case _ =>
+        base.listSortRows(isDescending, true)
```

Comment:
I just realize that this still has bug when the input is array of nested types having `-NaN`. The solution for this is to recursively applying `normalizeNANsAndZeros` on all children columns of type float/double. That can be done using `ColumnCastUtil.deepTransform`.

Reply:
I guess `GpuSortArray` haven't supported nested array (aka array of array). But array of struct is supported. (https://github.com/NVIDIA/spark-rapids/blob/branch-22.10/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala#L2771-L2779)

Could `ColumnCastUtil.deepTransform` be used on array of struct?

Reply:
It can be used for any type because it is "deep".

Reply:
> The solution for this is to recursively applying normalizeNANsAndZeros on all children columns of type float/double. That can be done using ColumnCastUtil.deepTransform.

As `Struct` is sorted by the first field, we should only apply `normalizeNansAndZeros` on the first child column of the struct. If `ColumnCastUtil.deepTransform` is recursively called on all the child arrays, it will introduce much overhead and might lead to a wrong result.

Reply:
> As Struct is sorted by the first field

No, structs are sorted by all fields. If the rows of the first field are equal, the rows of the second field will be compared and so on.

Overhead here is unavoidable if we want correctness.

Reply:
Updated.